src/main/java/uk/gov/hmcts/reform/sscs/services/ProcessingVenueMigrationService.java

```diff
@@ -0,0 +1,47 @@
+package uk.gov.hmcts.reform.sscs.services;
+
+import static uk.gov.hmcts.reform.sscs.util.MigrationStringUtils.decompressAndB64Decode;
+
+import java.io.IOException;
+import java.util.concurrent.atomic.AtomicInteger;
+import lombok.extern.slf4j.Slf4j;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.idam.IdamService;
+import uk.gov.hmcts.reform.sscs.services.ccd.CcdCasesSender;
+
+
+@Service
+@Slf4j
+public class ProcessingVenueMigrationService {
+
+    private final CcdCasesSender ccdCasesSender;
+    private final IdamService idamService;
+
+    @Value("${features.venue-migration.encoded-data-string}")
+    private String encodedDataString;
+
+    public ProcessingVenueMigrationService(CcdCasesSender ccdCasesSender, IdamService idamService) {
+        this.ccdCasesSender = ccdCasesSender;
+        this.idamService = idamService;
+    }
+
+    public void process(String venue) throws IOException {
```

Comment:
this is exactly the same class as the interpreter one, except the call to updateProcessingVenue right?
let's reuse the existing class in that case, and merge the updateProcessingVenue/updateLanguage methods